src/backend/wsl.ts

```diff
@@ -1014,6 +1012,7 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
         const prepActions = [(async() => {
           await this.ensureDistroRegistered();
           await this.initDataDistribution();
+          await this.upgradeDistroAsNeeded();
```

Comment:
`initDataDistribution` needs to the the upgraded distro registered, so that it can copy files from there into the data distribution.